src/api/OffsetBasedAPI.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @flow
+ * @file class for Box offset based API's to inherit common functionality from
+ * @author Box
+ */
+import Base from './Base';
+
+type Params = {
+    offset: number,
+    limit: number,
+    fields?: string
+};
+
+type Data = {
+    entries: Array<any>,
+    total_count: number
+};
+
+class OffsetBasedApi extends Base {
+    /**
+     * @property {Data}
+     */
+    data: Data;
+
+    /**
+     * Gets query params for the API
+     *
+     * @return the query params object
+     */
+    getQueryParameters(offset: number, limit: number, fields: ?Array<string>): Object {
+        const queryParams: Params = {
+            offset,
+            limit
+        };
+
+        if (fields && fields.length > 0) {
+            queryParams.fields = fields.toString();
+        }
+
+        return queryParams;
+    }
+
+    /**
+     * Determines if the API has more items to fetch
+     * @param {number} offset the offset from the start to start fetching at
+     * @param {number} totalCount the total number of items
+     * @return {boolean} true if there are more items
+     */
+    hasMoreItems(offset: number, totalCount?: number): boolean {
```

Comment:
Double check if you aren't falling for the same gotcha about total count issue I did that caused this patchset
https://github.com/box/box-ui-elements/commit/2980e0ca42e08194c3c3ab51c4268a775cf6e8cf

Reply:
Based on my understanding, I think my implementation is correct here